bndtools.core/src/bndtools/editor/project/RunRequirementsPart.java

```diff
@@ -257,8 +259,8 @@ private void doAddBundle() {
             WizardDialog dialog = new WizardDialog(getSection().getShell(), wizard);
 
             if (Window.OK == dialog.open()) {
-                List<VersionedClause> result = wizard.getSelectedBundles();
-                List<Requirement> adding = new ArrayList<Requirement>(result.size());
+                final List<VersionedClause> result = wizard.getSelectedBundles();
+                final List<Requirement> adding = new ArrayList<Requirement>(result.size());
```

Comment:
This depends on what you would like to display to the user:

If we dont want to display the duplicates to the user in the dialog, I can make this a Set. I decided to go for separate lists in order to give some feedback. Even though the dialog is not pretty and might anoy somebody I thought it to be better than a user who is wondering why nothing happened (editor marked dirty) after he added bundles.

Best solution might be to check if a bundle can be added during the selection but this has to be done in multiple places, so I went for the approach with the smallest impact.